### PR TITLE
feat(telecom): add inward postcode for uk phone number

### PR DESCRIPTION
**Pull request by qpavy**
ref: MANAGER-11693

| Question         | Answer
| ---------------- | ---
| Branch?          | `master`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #MANAGER-11693
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- ~~[ ] Breaking change is mentioned in relevant commits~~

## Related

- [ ] TRANS-52539

### CODE REVIEW

**Review by darsene — CHANGES_REQUESTED** [review]
replace `isServiceInUK` function with property instead

File: packages/manager/apps/telecom/src/app/telecom/telephony/service/contact/contact.html [hunk 1]
```diff
@@ -234,34 +234,73 @@
...
+                                    error-messages="{ 'required': ('telephony_service_contact_must_fill' | translate), 'pattern': ('telephony_service_contact_bad_inward' | translate) }"
+                                    class="col-md-4"
+                                >
+                                    <input
+                                        type="text"
+                                        id="contactInwardPostCode"
+                                        name="contactInwardPostCode"
+                                        class="oui-input"
+                                        required
+                                        data-ng-pattern="/^\d[a-zA-Z]{2}$/"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/service/contact/contact.html:298]: Can you move the regex in constant file.

File: packages/manager/apps/telecom/src/app/telecom/telephony/service/contact/contact.html [hunk 2]
```diff
@@ -234,34 +234,73 @@
...
+                                    <oui-select
+                                        items="$ctrl.autoCompletePostCode"
+                                        model="$ctrl.ukPostCode.outwardPostCode"
+                                        name="contactOutwardPostCode"
+                                        id="contactOutwardPostCode"
+                                        data-on-change="$ctrl.onUKPostCodeChange()"
+                                        required
+                                        searchable
+                                    >
+                                    </oui-select>
```
[the remark this answers is not in the document]
Comment (qpavy) [reply packages/manager/apps/telecom/src/app/telecom/telephony/service/contact/contact.html]: I add an input text field as it is possible to have non geographic number in uk

File: packages/manager/apps/telecom/src/app/telecom/telephony/service/contact/contact.html [hunk 3]
```diff
@@ -234,34 +234,85 @@
...
+                                        name="contactPostCode"
+                                        class="oui-input"
+                                        required
+                                        data-ng-change="$ctrl.onPostCodeChange()"
+                                        data-ng-model="$ctrl.directoryForm.postCode"
+                                        data-ng-model-options="{ debounce: 500 }"
+                                    />
+                                    <oui-select
+                                        data-ng-if="$ctrl.autoCompletePostCode.length > 0"
+                                        items="$ctrl.isServiceInUKautoCompletePostCode"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/service/contact/contact.html:258]:
```suggestion
                                        items="$ctrl.autoCompletePostCode"
```

File: packages/manager/apps/telecom/src/app/telecom/telephony/service/contact/contact.html [hunk 4]
```diff
@@ -234,34 +234,85 @@
...
+                                        searchable
+                                    >
+                                    </oui-select>
+                                </oui-field>
+
+                                <oui-field
+                                    data-ng-if="$ctrl.isServiceInUK"
+                                    label="{{:: 'telephony_service_contact_outwardcode' | translate }}*"
+                                    error-messages="{ 'required': ('telephony_service_contact_must_fill' | translate) }"
+                                    data-ng-class="$ctrl.isServiceInUK ? 'col-md-4' :  'col-md-12'"
```
Comment (ghyenne) [inline packages/manager/apps/telecom/src/app/telecom/telephony/service/contact/contact.html:273]:
```suggestion
                                    data-ng-if="$ctrl.isServiceInUK"
                                    label="{{:: 'telephony_service_contact_outwardcode' | translate }}*"
                                    error-messages="{ 'required': ('telephony_service_contact_must_fill' | translate) }"
                                    class="col-md-4"
```

you are data-ng-if so this contition on class isn't required

**Review by darsene — DISMISSED** [review]

**Review by ghyenne — APPROVED** [review]

**Review by darsene — DISMISSED** [review]

**Review by anooparveti — APPROVED** [review]

**Pull request merged**